front/app/containers/EmailChange/index.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState } from 'react';
+
+// components
+import { Box } from '@citizenlab/cl2-component-library';
+import { Helmet } from 'react-helmet';
+import { StyledContentContainer } from 'components/smallForm';
+import GoBackButton from 'components/UI/GoBackButton';
+import EmailConfirmation from 'containers/NewAuthModal/steps/EmailConfirmation';
+import Modal from 'components/UI/Modal';
+import Error from 'components/UI/Error';
+import CancelUpdate from './CancelUpdate';
+import UpdateEmailForm from './UpdateEmailForm';
+
+// api
+import clHistory from 'utils/cl-router/history';
+import useAuthUser from 'hooks/useAuthUser';
+import confirmEmail from 'api/authentication/confirmEmail';
+
+// hook form
+import { object, string } from 'yup';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+// intl
+import { useIntl } from 'utils/cl-intl';
+import messages from './messages';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+
+// typings
+import { Status, ErrorCode } from 'containers/NewAuthModal/typings';
+import { ERROR_CODE_MESSAGES } from 'containers/NewAuthModal';
+
+export type FormValues = {
+  email: string;
+};
+
+const EmailChange = () => {
+  const { formatMessage } = useIntl();
+  const authUser = useAuthUser();
+  const [openConfirmationModal, setOpenConfirmationModal] = useState(false);
+  const [confirmationStatus, setConfirmationStatus] = useState<Status>('ok');
+  const [confirmationError, setConfirmationError] = useState<ErrorCode | null>(
+    null
+  );
+  const [updateSuccessful, setUpdateSuccessful] = useState(false);
+  const [updateCancelled, setUpdateCancelled] = useState(false);
+
+  const schema = object({
+    email: string()
+      .email(formatMessage(messages.emailInvalidError))
+      .required(formatMessage(messages.emailEmptyError)),
+  });
+
+  const methods = useForm<FormValues>({
+    mode: 'onBlur',
+    defaultValues: {
+      email: '',
+    },
+    resolver: yupResolver(schema),
+  });
+
+  // Once auth user is fetched, set the email field to the user's email
+  useEffect(() => {
+    if (!isNilOrError(authUser) && authUser.attributes.email) {
+      if (!methods.watch('email')) {
+        methods.setValue('email', authUser.attributes.email);
+      }
+    }
+  }, [authUser, methods]);
```

Comment:
The issue was that the default value wasn't being set to the existing email, since the authUser hadn't loaded in time when the default value was being set in "methods". I've used a useEffect instead to handle it, but maybe there is a better way?